### PR TITLE
[WIP] update backend to PyMC v4

### PR DESCRIPTION
**Pull request by aloctavodia**

### CODE REVIEW

**Comment by canyon289** [comment]
Maybe we should disable the linters for now so we can see if tests pass. Im happy to help out as well

**Comment by aloctavodia** [comment]
Many test are still not passing

**Comment by canyon289** [comment]
Thats fine, if we skip linting we can see which tests fail

**Comment by tomicapretto** [comment]
I'm running a couple of models and getting this

```
/home/tomas/anaconda3/envs/bambiv4/lib/python3.9/site-packages/pymc/aesaraf.py:1005: UserWarning: The parameter 'updates' of aesara.function() expects an OrderedDict, got <class 'dict'>. Using a standard dictionary here results in non-deterministic behavior. You should use an OrderedDict if you are using Python 2.7 (collections.OrderedDict for older python), or use a list of (shared, update) pairs. Do not just convert your dictionary to this type before the call as the conversion will still be non-deterministic.
```

Are you familiar with it?

**Comment by canyon289** [comment]
for the dictionary one yes but newer python dicts are ordered by default I believe. we may be able to change that in aesara. I can check later

File: bambi/tests/test_built_models.py [hunk 1]
```diff
@@ -377,6 +376,7 @@ def dicts_close(a, b):
     assert all([dicts_close(priors0[x], priors1[x]) for x in priors0.keys()])
 
 
+@pytest.mark.skip(reason="Standard deviation results are different and I don't know why")
```
Comment (tomicapretto) [inline bambi/tests/test_built_models.py:379]: I'm adding this line to skip this test. There's a difference in the standard deviation. I tried to fix it on my side but I'm not familiar with how this works. I guess that PyMC changed something in its `find_MAP` or `find_hessian` and that's affecting this comparison.

**Comment by tomicapretto** [comment]
Some of our tests are failing because `.squeeze()` does not seem to work as before on tensor variables.

See the following example

```python
import aesara
import pymc as pm

coords = {
    "A": list("abc"), 
    "B": list("x")
}

with pm.Model(coords=coords) as pm_model:
    b = pm.Normal("b", dims=("A", "B"))
    c = b.squeeze()
    
print(b.shape.eval())
print(c.shape.eval()) # Expect this to be just 3

print("Aesara version", aesara.__version__)
print("PyMC version", pm.__version__)
```

```
[3 1]
[3 1]
Aesara version 2.5.1
PyMC version 4.0.0b6
```

I'm going to check with Ricardo if this is intended or unintended behavior.

**Comment by aloctavodia** [comment]
The Op i1 error is also something that should be fixed on the aesara side.

**Comment by tomicapretto** [comment]
Ricardo responded this is because of PyMC, not Aesara. And it should be working when https://github.com/pymc-devs/pymc/issues/5751 is addressed. In the meantime, we could fix it on our side without relying on `.squeeze()` but I'm not sure if we want to do it.

**Comment by aloctavodia** [comment]
I think it makes sense to wait.

**Comment by aloctavodia** [comment]
Yes we merge and then we keep updating the v4 branch until is ready.

**Comment by tomicapretto** [comment]
Merging now. This will be in `v4` branch.

**Pull request merged**